chorus_ui/src/ts/UbiClient.ts

```diff
@@ -70,30 +77,62 @@ export class UbiLogger {
             withCredentials:true
         });
 
-        //TODO: init if it doesn't exist
-        //this.init();
+
+        //if the ubi store doesn't exist, create it
+        if(this.get_stores().indexOf(this.ubi_store) != -1){
+            this.init();
+        }
+
+    }
+
+    /**
+     * 
+     * @returns All available Ubi stores
+     */
+    get_stores(){
+        const response = this._get(this.url).then(
+        (response) => {
+            if(response.status == 200){
+                console.log('Listing stores: ' + JSON.stringify(response));
+                return response['data']['stores'];
+            }
+            }
+        ).catch(
+            (error) => {
+                console.warn('Error querying stores: ' + error);
+                console.warn('Ubi Store ' + this.ubi_store + ' needs to be initialized');
+                } 
+        )
+    
+        return []
     }
 
+    /**
```

Comment:
Fine this is in here, but I don't expect the store to be initialized from the Javascript. Whoever is installing the plugin in OpenSearch will then initialize the store. Just wasn't sure if this is for reference, or if we are encouraging users to initialize it through Javascript.